source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -594,9 +594,37 @@ namespace
         return destNode;
     }
 
-    void writeMaterialElements(const PvtNode* node, NodePtr mxNode, const string& nodeName, DocumentPtr doc, const RtWriteOptions* writeOptions)
+    void createMaterialNode(PvtNode* node, NodePtr mxNode, DocumentPtr doc)
     {
-        MaterialPtr material = doc->addMaterial(mxNode->getName() + "_Material");
+        // Check to see if the surfaceshader node is already connected to a material node
+        size_t numOutputs = node->numOutputs();
+        bool isConnectedToMaterialNode = false;
+        for (size_t i=0; i<numOutputs; ++i)
+        {
+            RtPort outputPort = node->getPort(node->getOutputsOffset() + i);
+            size_t numDestinationPorts = outputPort.numDestinationPorts();
+            for (size_t j=0; j<numDestinationPorts; ++j)
+            {
+                RtNode connectedNode(outputPort.getDestinationPort(j).getNode());
+                if (connectedNode.numOutputs() > 0 &&
+                    connectedNode.getOutput(0).getType() == "material")
+                {
+                    isConnectedToMaterialNode = true;
+                    break;
+                }
+            }
+        }
+        
+        if (!isConnectedToMaterialNode)
+        {
+            NodePtr materialNode = doc->addNode("surfacematerial", mxNode->getName() + "_SurfaceMaterial", "material");
+            materialNode->setConnectedNode("surfaceshader", mxNode);
+        }
+    }
+
+    void writeMaterialElementsHelper(PvtNode* node, NodePtr mxNode, const string& materialBaseName, const string& nodeName, DocumentPtr doc, const RtWriteOptions* writeOptions)
```

Comment:
Helper routine that produces the material element associated with a surface shader. Has the name of either the surface shader or respective material node as it's base.